cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -138,22 +135,33 @@ public Count addRequestAndGetCount(ProgramRunId programRunId) throws Exception {
    */
   public void addRequest(ProgramRunId programRunId) {
```

Comment:
nit: would it be better to name it `addLaunchingRequest` ? similar comments to `addRequestAndGetCount` and `removeRequest`

Reply:
This is a great idea. The main reason I didn't do it is `addLaunchingRequestAndGetCount` becomes quite long.